### ISSUE TITLE
Guns (esp. coilgun and chaingun) don't play firing sounds once in a few shots (audio buffer problem?)

### ISSUE THREAD

**Issue by Krzeszny** [Regalis11/Barotrauma#9489]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

_This has been bothering me for over a year now._ 

Coilgun and chaingun can miss up to **3** firing sounds in a row. 
This only starts happening after firing the first 15 or so shots.

https://user-images.githubusercontent.com/10529134/176560108-b07ce225-9ec0-4b9b-a7a3-06397f831924.mp4

https://user-images.githubusercontent.com/10529134/176560124-67414964-5a09-4e05-9184-a340111c3579.mp4

### Reproduction steps

1. Start any game
2. Start firing a gun continuously
3. Listen to 1-3 missing shot audio once every 2-10 shots.

### Bug prevalence

Happens every time I play

### Version

at least from 0.14 up to 0.18.13.0

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/91e20c93ea89d5eb7a7ec4833f3525869e0b4d04

**Comment by NilanthAnimosus** [comment]
Tested against bugfixes commit https://github.com/Regalis11/Barotrauma-development/commit/6b71782a30a11a63291d62eee5d52df2371e8ad9

Without any modifications to weapons (Testing with a Winterhalter class submarine) I was unable to get the sound to skip using coilguns and chainguns. However when upgrading to max (+50% fire rate) I was able to get only the coilguns to skip, and it felt like 1 in every 50+ or more shots a singular skip and more likely tied down to how many other background noises are going on.

Don't think this needs any further attention to given your unlikely to skip a sound before emptying almost a full ammunition box from my experience now - at least when in an environment where not too many other noises are occurring.

the real case this is likely to skip now though is if multiple coilgun's, or a coilgun with lots of creature noises, footsteps, flooding and other such to all add up at the same time saturating the cap on simultaneous sounds that can play, though not much can be done about that. Can close if this is considered fine given I can just barely reach skipping audio without much going on I think the above scenario could still be feasible in busy times.
